egs2/snips/asr1/local/data.sh

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env bash
+
+# Copyright 2021 Yuekai Zhang
+#  Apache 2.0  (http://www.apache.org/licenses/LICENSE-2.0)
+
+. ./path.sh || exit 1;
+. ./cmd.sh || exit 1;
+. ./db.sh || exit 1;
+
+# general configuration
+stage=1       # start from 0 if you need to start from data preparation
+stop_stage=3
+SECONDS=0
+
+log() {
+    local fname=${BASH_SOURCE[1]##*/}
+    echo -e "$(date '+%Y-%m-%dT%H:%M:%S') (${fname}:${BASH_LINENO[0]}:${FUNCNAME[1]}) $*"
+}
+
+
+# Set bash to 'debug' mode, it will exit on :
+# -e 'error', -u 'undefined variable', -o ... 'error in pipeline', -x 'print commands',
+set -e
+set -u
+set -o pipefail
+
+. utils/parse_options.sh
+
+log "data preparation started"
+
+if [ -z "${SNIPS}" ]; then
+    log "Fill the value of 'SNIPS' of db.sh"
+    exit 1
+fi
+
+if [ ${stage} -le 0 ] && [ ${stop_stage} -ge 0 ]; then
+    log "stage 0: Download Data to"
+    local/download_and_untar.sh 
+fi
+
+if [ ${stage} -le 1 ] && [ ${stop_stage} -ge 1 ]; then
+    log "stage 1: Preparing Data for snips"
+    mkdir -p data
+    python local/data_prep.py --wav_path $SNIPS
+fi
+
+if [ ${stage} -le 2 ] && [ ${stop_stage} -ge 2 ]; then
+   log "stage 2: Processing the text files"
+   mkdir -p data/tmp
+   sort -u data/text.trans > data/tmp/text.sort
+   sort  -o data/wav.scp  data/wav.scp
+   sort -uo data/non_linguistic_symbols.txt data/non_linguistic_symbols.txt
+   sort -u data/semantics > data/tmp/semantics.sort
+   rm -r data/semantics
+   awk '{print $(NF)}' data/tmp/text.sort > data/tmp/wavs
+   paste -d "|" data/tmp/wavs data/tmp/semantics.sort > data/tmp/wav_sem_text
+   cut -d "|" -f 1,3 data/tmp/wav_sem_text > data/tmp/wav_sem
+   sed -r 's/\|/ /g' data/tmp/wav_sem > data/text
+   mkdir -p data/all
+   mv data/utt2spk data/text data/wav.scp data/all 
+   utils/fix_data_dir.sh data/all
+   rm -r data/tmp
+fi
+
+if [ ${stage} -le 3 ] && [ ${stop_stage} -ge 3 ]; then
```

Comment:
Could you make sure that you are using the splits as used in "Tie your embeddings down: Cross-modal
latent spaces for end-to-end spoken language understanding," since those are the splits that have been used to benchmark SLU models.

Reply:
https://github.com/alexa/alexa-end-to-end-slu/blob/0d4a44edd468d9951ac9f4d0917585d3113171b2/dataprep/prepare_snips.py#L23 I would say the dataset I downloaded from snips' email has the different structure with the Amazon paper. I didn't have any .csv file. I think it is hard to know the order of utterances in the Amazon paper.

Reply:
Hi I am sorry I was not aware the complete.csv is no longer part of Snips dataset. The collaborators with whom I worked earlier used metadata.json to create complete.csv. I think we should definitely make sure that train:valid: test split is 0.8:0.1:0.1 so as to ensure train:valid:test split size is similar to the ones used in prior work.